### ISSUE TITLE
Bundler 1.12.5 does not appear to respect `ruby RUBY_VERSION` for dependencies of gems in Gemfile

### ISSUE THREAD

**Issue by ferventcoder** [bundler/bundler#4819]
I'm not sure if I'm doing something wrong here, if this is a bug or is by design.

Gemfile:

~~~ruby
ruby RUBY_VERSION
source ENV['GEM_SOURCE'] || 'https://rubygems.org'

gem 'puppet', :require => false
~~~

~~~sh
$ rbenv shell 1.9.3-p392
$ ruby --version
ruby 1.9.3p392 (2013-02-22 revision 39386) [x86_64-darwin12.3.0]
$ bundle --version
Bundler version 1.12.5
$ bundle install
Fetching gem metadata from https://rubygems.org/
Fetching version metadata from https://rubygems.org/
Resolving dependencies...
Using CFPropertyList 2.2.8
Using bundler 1.12.5
Installing json_pure 2.0.2

Gem::InstallError: json_pure requires Ruby version ~> 2.0.
Installing facter 2.4.6
An error occurred while installing json_pure (2.0.2), and Bundler cannot continue.
Make sure that `gem install json_pure -v '2.0.2'` succeeds before bundling.
~~~

**Comment by ferventcoder** [comment]
Discovered as part of https://github.com/bundler/bundler/pull/4651#issuecomment-235382389

**Comment by ferventcoder** [comment]
Is this a bug or by design?

**Comment by segiddins** [comment]
Can you check if this is fixed on master? I've recently fixed several bugs in that area. Thanks!

**Comment by lamont-granquist** [comment]
I'm seeing this fixed on master

**Comment by ferventcoder** [comment]
Thanks @lamont-granquist - I haven't gotten back to test this yet. 👍

**Comment by lamont-granquist** [comment]
Would definitely be good to see this getting released, everyone is deprecating 2.1.x support so we're getting a plague of failing bundle installs, and this would help some.

**Comment by lynnco** [comment]
The relevant changes were released in 1.13, closing
